src/xviewer-list-store.c

```diff
@@ -627,9 +644,55 @@ xviewer_list_store_add_files (XviewerListStore *store, GList *file_list)
 			g_object_unref (file);
 		} else if (file_type == G_FILE_TYPE_REGULAR && !singleton_list) {
 			xviewer_list_store_append_image_from_file (store, file);
+
+			file = g_file_get_parent (file);
```

Comment:
`file` needs to be `g_object_unref`'d here before re-assigning it.

Reply:
Unref of file added before L648. I'll work through the remaining points and then resubmit the file.